webapp/src/js/services/rules-engine.js

```diff
@@ -187,8 +187,7 @@ var nools = require('nools'),
           key: 'rules-engine',
           callback: updateFacts,
           filter: function(change) {
-            return change.doc.form ||
-                   ContactSchema.getTypes().indexOf(change.doc.type) !== -1;
+            return change.doc.form || ContactTypes.includes().indexOf(change.doc);
```

Comment:
This throws an error cause `ContactTypes.includes` is:
```
includes: doc => {
  return doc.type === 'contact' ||  // configurable hierarchy
             isHardcodedType(doc.type); // hardcoded hierarchy
  }
```

and `doc` is undefined.

Reply:
:man_facepalming: